File: frontend/src/types/template.ts
export type VariableType = "text" | "date" | "number" | "currency" | "select";

export interface Variable {
  key: string;
  label: string;
  type: VariableType;
  required: boolean;
  options?: string[];
  description?: string;
}

export interface Template {
  id: string;
  name: string;
  category: string;
  description: string;
  jurisdiction?: string;
  variables: Variable[];
  content: string;
}

export interface TemplateIndexEntry {
  id: string;
  name: string;
  category: string;
  description: string;
  file: string;
}

File: backend/app/models/template.py
from typing import Literal, Optional
from pydantic import BaseModel


class Variable(BaseModel):
    key: str
    label: str
    type: Literal["text", "date", "number", "currency", "select"]
    required: bool = True
    options: Optional[list[str]] = None
    description: Optional[str] = None


class Template(BaseModel):
    id: str
    name: str
    category: str
    description: str
    jurisdiction: Optional[str] = None
    variables: list[Variable]
    content: str


class TemplateIndexEntry(BaseModel):
    id: str
    name: str
    category: str
    description: str
    file: str
